ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -92,15 +104,57 @@ func ConvertToTaskDefinition(taskDefinitionName string, context *project.Context
 		containerDefinitions = append(containerDefinitions, containerDef)
 	}
 
+	var ecsFields *ECSFields
+	if ecsFieldsFileName != "" {
+		data, err := readECSFields(ecsFieldsFileName)
+		ecsFields = data
```

Comment:
Why no assign ecsFields directly? `ecsFields, err := readECSFields(ecsFieldsFileName)

Reply:
Because it doesn't compile:

```
diff --git a/ecs-cli/modules/utils/compose/convert_task_definition.go b/ecs-cli/modules/utils/compose/convert_task_definition.go
index 50d427d..93df616 100644
--- a/ecs-cli/modules/utils/compose/convert_task_definition.go
+++ b/ecs-cli/modules/utils/compose/convert_task_definition.go
@@ -106,8 +106,8 @@ func ConvertToTaskDefinition(taskDefinitionName string, context *project.Context
 
        var ecsFields *ECSFields
        if ecsFieldsFileName != "" {
-               data, err := readECSFields(ecsFieldsFileName)
-               ecsFields = data
+               ecsFields, err := readECSFields(ecsFieldsFileName)
+               // ecsFields = data
                if err != nil {
                        return nil, err
                }

~/go/src/github.com/aws/amazon-ecs-cli(support_ecs_fields)$ make
./scripts/build_binary.sh ./bin/local
# github.com/aws/amazon-ecs-cli/ecs-cli/modules/utils/compose
ecs-cli/modules/utils/compose/convert_task_definition.go:109: ecsFields declared and not used
make: *** [bin/local/ecs-cli] Error 2
```

I didn't like this either, and it feels clunky. Not sure if there's a better way to do it. @aaithal or @samuelkarp ?

Reply:
So I haven't actually read any of the code and I'm just responding to the comment here + diff, but I'd probably do it like this:

```go
var ecsFields *ECSFields
if ecsFieldsFileName != "" {
	var err error
	ecsFields, err = readECSFields(ecsFieldsFileName)
	if err != nil {
		return nil, err
	}
}
```

This avoids accidental shadowing that happens during the implicit declaration with `:=`; instead we have explicit declaration with `var` and simple assignment with `=`.

Reply:
@samuelkarp I tried that too, but I get an error because `err` is not instantiated.

```
~/go/src/github.com/aws/amazon-ecs-cli(support_ecs_fields)$ gd
diff --git a/ecs-cli/modules/utils/compose/convert_task_definition.go b/ecs-cli/modules/utils/compose/convert_task_definition.go
index efd2003..7cec657 100644
--- a/ecs-cli/modules/utils/compose/convert_task_definition.go
+++ b/ecs-cli/modules/utils/compose/convert_task_definition.go
@@ -106,8 +106,7 @@ func ConvertToTaskDefinition(taskDefinitionName string, context *project.Context
 
        var ecsParams *ECSParams
        if ecsParamsFileName != "" {
-               data, err := readECSParams(ecsParamsFileName)
-               ecsParams = data
+               ecsParams, err = readECSParams(ecsParamsFileName)
                if err != nil {
                        return nil, err
                }

~/go/src/github.com/aws/amazon-ecs-cli(support_ecs_fields)$ make
./scripts/build_binary.sh ./bin/local
# github.com/aws/amazon-ecs-cli/ecs-cli/modules/utils/compose
ecs-cli/modules/utils/compose/convert_task_definition.go:109: undefined: err
ecs-cli/modules/utils/compose/convert_task_definition.go:110: undefined: err
ecs-cli/modules/utils/compose/convert_task_definition.go:111: undefined: err
```

Reply:
You missed a line from my diff; you need the `var err error` to declare the variable.

Reply:
Oh, oops. Ended up changing this all anyway, but thanks for your wisdom :)